addons/binding/org.openhab.binding.plugwise/src/main/java/org/openhab/binding/plugwise/internal/protocol/AnnounceAwakeRequestMessage.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.plugwise.internal.protocol;
+
+import static org.openhab.binding.plugwise.internal.protocol.field.MessageType.ANNOUNCE_AWAKE_REQUEST;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.openhab.binding.plugwise.internal.protocol.field.MACAddress;
+
+/**
+ * A sleeping end device (SED: Scan, Sense, Switch) sends this message to announce that is awake.
+ *
+ * @author Wouter Born - Initial contribution
+ */
+public class AnnounceAwakeRequestMessage extends Message {
+
+    public enum AwakeReason {
+
+        /** The SED joins the network for maintenance */
+        MAINTENANCE(0),
+
+        /** The SED joins a network for the first time */
+        JOIN_NETWORK(1),
+
+        /** The SED joins a network it has already joined, e.g. after reinserting a battery */
+        REJOIN_NETWORK(2),
+
+        /** When a SED switches a device group or when reporting values such as temperature/humidity */
+        NORMAL(3),
+
+        /** A human pressed the button on a SED to wake it up */
+        WAKEUP_BUTTON(5);
+
+        private static final Map<Integer, AwakeReason> REASONS_BY_VALUE = new HashMap<>();
+
+        static {
+            for (AwakeReason type : AwakeReason.values()) {
```

Comment:
Yes it was a bit overkill for this class. So I've replaced it with some lamba logic.